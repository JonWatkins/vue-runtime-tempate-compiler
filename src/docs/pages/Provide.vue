<template>
  <div class="page">
    <h1 id="top" class="mb-3">
      <router-link to="/guide/provide#top" class="header-anchor">
        <span class="header-anchor--hash">#</span>
        Provide
      </router-link>
    </h1>

    <p>
      Just like the <code>props</code> the <code>provide</code> properties are
      available in the <code>vue-runtime-template-compiler</code>. The same as
      the other parent property you use them as if you are writing bindings in
      the parent's compiled template.
    </p>

    <pre class="language-html mb-4"><code>{{ example | trim }}</code></pre>

    <PageNav>
      <template v-slot:left>
        <router-link to="/guide/props">
          <i class="fas fa-arrow-left"></i>
          Props
        </router-link>
      </template>
      <template v-slot:right>
        <router-link to="/guide/filters">
          Filters
          <i class="fas fa-arrow-right"></i>
        </router-link>
      </template>
    </PageNav>
  </div>
</template>

<script>
import PageNav from '../components/PageNav.vue'

const example = `
<template>
  <runtime-template-compiler :template="template" \/>
<\/template>

<script>
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  provide: {
    name: 'Bob'
  }
  data: () => ({
    template: \`<div>{{ name }}!<\/div>\`,
  })
}
<\/script>
`

export default {
  name: 'Provide',

  components: {
    PageNav
  },

  data() {
    return {
      example
    }
  }
}
</script>
