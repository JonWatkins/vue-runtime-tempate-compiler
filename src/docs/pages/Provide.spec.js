import { shallowMount } from '@vue/test-utils'
import Provide from './Provide.vue'
const { localVueInst } = global

let wrapper

describe('Provide', () => {
  beforeEach(() => {
    wrapper = shallowMount(Provide, {
      localVue: localVueInst
    })
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Provide')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
