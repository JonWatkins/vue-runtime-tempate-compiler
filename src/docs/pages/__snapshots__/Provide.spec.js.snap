// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provide render correctly 1`] = `
<div class="page">
  <h1 id="top" class="display-4 mb-3">
    <router-link-stub to="/guide/provide#top" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      Provide
    </router-link-stub>
  </h1>
  <p>
    Just like the <code>props</code> the <code>provide</code> properties are
    available in the <code>vue-runtime-template-compiler</code>. The same as
    the other parent property you use them as if you are writing bindings in
    the parent's compiled template.
  </p> <pre class="language-html mb-4"><code>&lt;template&gt;
  &lt;runtime-template-compiler :template="template" /&gt;
&lt;/template&gt;

&lt;script&gt;
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  provide: {
    name: 'Bob'
  }
  data: () =&gt; ({
    template: \`&lt;div&gt;{{ name }}!&lt;/div&gt;\`,
  })
}
&lt;/script&gt;</code></pre>
  <pagenav-stub></pagenav-stub>
</div>
`;
